src/bytecode/flags.py

```diff
@@ -80,19 +108,7 @@ def infer_flags(
         if isinstance(bytecode, _bytecode.ControlFlowGraph)
         else bytecode
     )
-    instr_names = {
-        i.name
-        for i in instructions
-        if not isinstance(
-            i,
-            (
-                _bytecode.SetLineno,
-                _bytecode.Label,
-                _bytecode.TryBegin,
-                _bytecode.TryEnd,
-            ),
-        )
-    }
+    instr_names = {i.name for i in instructions if not ignored_instruction(i)}
```

Comment:
Could you make it so we only iterate through the instructions once ? by collecting the names and the fact that we know we are dealing with a generator in the same pass.